src/EventStore.Core/Services/Storage/StorageReaderService.cs

```diff
@@ -35,18 +35,23 @@ public StorageReaderService(IPublisher bus, ISubscriber subscriber, IReadIndex r
             _readIndex = readIndex;
             _threadCount = threadCount;
 
-            var readerWorker = new StorageReaderWorker(bus, readIndex, writerCheckpoint);
-            var storageReaderBus = new InMemoryBus("StorageReaderBus", watchSlowMsg: false);
-            storageReaderBus.Subscribe<ClientMessage.ReadEvent>(readerWorker);
-            storageReaderBus.Subscribe<ClientMessage.ReadStreamEventsBackward>(readerWorker);
-            storageReaderBus.Subscribe<ClientMessage.ReadStreamEventsForward>(readerWorker);
-            storageReaderBus.Subscribe<ClientMessage.ReadAllEventsForward>(readerWorker);
-            storageReaderBus.Subscribe<ClientMessage.ReadAllEventsBackward>(readerWorker);
-            storageReaderBus.Subscribe<StorageMessage.CheckStreamAccess>(readerWorker);
+            StorageReaderWorker[] readerWorkers  = new StorageReaderWorker[threadCount];
+            InMemoryBus[] storageReaderBuses = new InMemoryBus[threadCount];
```

Comment:
@shaan1337  Can you please explain the need of introducing an array of InMemoryBus here?

Reply:
@riccardone thanks for pointing this to my attention, I'll have a look to see if we can reuse the same object but at first sight, it looks like it's saving some internal state

Reply:
@riccardone we need an array of InMemoryBus here (one per thread) because the internal state of the StorageReaderWorker is changing after each message is published.